### ISSUE TITLE
Minor fixes to split-paired-reads.py

### ISSUE THREAD

**Issue by ctb** [dib-lab/khmer#763]
Once #762 is merged, there are still a few updates to split-paired-reads that could usefully be done.

First, to support https://github.com/ged-lab/khmer/issues/732#issuecomment-71903726 we need to default to stdin.  That can be done by adding ``nargs='?', default='/dev/stdin'`` to line 60, although we also need to require -1 and -2 options in that case.

Second, we should switch to using the broken_paired_reader in khmer.utils that #759 will introduce.

Third, once we can read broken-paired reads, we should allow users to override the requirement that reads be perfectly interleaved. (By default, we should require perfect interleaving, though, because it's the main use case.)

**Comment by ctb** [comment]
Forgot to say - switching to broken_paired_reader is important because we will then support Casava 1.8 pair checking in split-paired-reads (ref #23).

**Comment by ctb** [comment]
Switched to broken_paired_reader & addressed third point in #818.  First point (supporting streaming) still needs to be done.

**Comment by mr-c** [comment]
@ctb This should support streaming as is with `/dev/stdin` being used as the input file name.

**Comment by mr-c** [comment]
@ctb I think this issue is finished; can you confirm?

**Comment by ctb** [comment]
No, not yet --

- [ ] 'infile' should default to /dev/stdin, nargs='?', and in that case -1 and -2 are required

**Comment by mr-c** [comment]
@ctb I don't think we can change that sort of default without bumping the version number?

**Comment by ctb** [comment]
Ok, but most of it can be done wo changing default; does specifying - work?

--
Titus Brown, ctbrown@ucdavis.edu

**Comment by mr-c** [comment]
@ctb '-' will once @anotherthomas is done with #960 (which adds this behaviour to screed)

**Comment by ctb** [comment]
This is now straightforward to do, but I continue to note that when /dev/stdin is specified the default filenames are broken ;).

**Comment by mr-c** [comment]
- [ ] don't break if '/dev/stdin' is the input filename

**Comment by ctb** [comment]
@SherineAwad are you working on this?

**Comment by SherineAwad** [comment]
@ctb just started.

**Comment by ctb** [comment]
ok - please create a new PR, thanks!

**Comment by ctb** [comment]
Fixed in #763 and elsewhere.

**Comment by mr-c** [comment]
@ctb #763 is fixed in #763? :-)

**Comment by ctb** [comment]
;)
